examples/combined_ODNP.py

```diff
@@ -142,121 +128,102 @@
-logger.debug("shape of saved enhancement data", psp.ndshape(DNP_data))
+logger.debug("shape of saved enhancement data", ndshape(DNP_data))
 # }}}
 # {{{run IR
 with power_control() as p:
     retval_IR = p.dip_lock(
-        uw_dip_center_GHz - uw_dip_width_GHz / 2,
-        uw_dip_center_GHz + uw_dip_width_GHz / 2,
+        parser_dict['uw_dip_center_GHz'] - parser_dict['uw_dip_width_GHz'] / 2,
+        parser_dict['uw_dip_center_GHz'] + parser_dict['uw_dip_width_GHz'] / 2,
     )
     p.mw_off()
     ini_time = time.time()  # needed b/c data object doesn't exist yet
     vd_data = run_IR(
         nPoints=nPoints,
-        nEchoes=nEchoes,
+        nEchoes=parser_dict['nEchoes'],
         vd_list_us=vd_list_us,
-        nScans=nScans,
-        adcOffset=adcOffset,
-        carrierFreq_MHz=carrierFreq_MHz,
-        p90_us=p90_us,
-        tau_us=tau_us,
-        repetition=FIR_rd,
-        output_name=output_name,
-        SW_kHz=SW_kHz,
+        nScans=parser_dict['thermal_nScans'],
+        adcOffset=parser_dict['adc_offset'],
+        carrierFreq_MHz=parser_dict['carrierFreq_MHz'],
+        p90_us=parser_dict['p90_us'],
+        tau_us=parser_dict['tau_us'],
+        repetition=parser_dict['FIR_rep'],
+        output_name=filename,
+        SW_kHz=parser_dict['SW_kHz'],
         ph1_cyc=IR_ph1_cyc,
         ph2_cyc=IR_ph2_cyc,
         ret_data=None,
     )
     vd_data.set_prop("start_time", ini_time)
     vd_data.set_prop("stop_time", time.time())
-    meter_power = -999
-    acq_params = {
-        j: eval(j)
-        for j in dir()
-        if j
-        in [
-            "adcOffset",
-            "carrierFreq_MHz",
-            "amplitude",
-            "nScans",
-            "nEchoes",
-            "p90_us",
-            "deadtime_us",
-            "repetition_us",
-            "SW_kHz",
-            "nPoints",
-            "deblank_us",
-            "tau_us",
-            "MWfreq",
-            "acq_time_ms",
-            "meter_power",
-        ]
-    }
-    vd_data.set_prop("acq_params", acq_params)
-    vd_data.set_prop("postproc_type", IR_postproc)
+    vd_data.set_prop("acq_params", parser_dict.asdict())
+    vd_data.set_prop("postproc_type", "spincore_IR_v1")
     vd_data.name("FIR_noPower")
-    vd_data.chunk("t", ["ph1", "ph2", "t2"], [len(IR_ph1_cyc), len(IR_ph2_cyc), -1])
+    vd_data = vd_data['nScans',-1:]
+    vd_data.chunk("t", ["ph2", "ph1", "t2"], [len(IR_ph1_cyc), len(IR_ph2_cyc), -1])
     vd_data.setaxis("ph1", IR_ph1_cyc / 4)
     vd_data.setaxis("ph2", IR_ph2_cyc / 4)
-    # Need error handling (JF has posted something on this..)
-    vd_data.hdf5_write(myfilename)
+    vd_data.setaxis('nScans',r_[0:parser_dict['nScans']])
+    nodename = vd_data.name()
+    with h5py.File(
+        os.path.normpath(os.path.join(target_directory, f"{filename_out}")
+    ) as fp:
+        if nodename in fp.keys():
+            print("this nodename already exists, so I will call it temp")
+            vd_data.name("temp_noPower")
```

Comment:
if the nodename exists, rename it temp_noPower to differentiate it from the other nodes